### ISSUE TITLE
AttributeError: type object 'cryptography.hazmat.bindings._rust.x509.Sct' has no attribute 'register'

### ISSUE THREAD

**Issue by StevenLooman** [sigstore/gh-action-sigstore-python#159]
**Description**

When using this action, version v2.1.1, the following error occurs:
```
...
Building wheels for collected packages: grpclib
  Building wheel for grpclib (pyproject.toml): started
  Building wheel for grpclib (pyproject.toml): finished with status 'done'
  Created wheel for grpclib: filename=grpclib-0.4.7-py3-none-any.whl size=76241 sha256=be10b43b85f3b111c9b4672bb7de8664c9f1d65ce023241868ba532abf88a5f3
  Stored in directory: /home/runner/.cache/pip/wheels/05/c0/1c/3d807409d0c67efeab2949832ba409205b1b6fe03f739ae4c1
Successfully built grpclib
Installing collected packages: appdirs, typing-extensions, securesystemslib, python-dateutil, pygments, pycparser, mdurl, importlib_resources, hyperframe, hpack, dnspython, charset-normalizer, annotated-types, requests, pydantic-core, multidict, markdown-it-py, h2, email-validator, cffi, tuf, rich, pydantic, id, grpclib, cryptography, pyOpenSSL, betterproto, sigstore-rekor-types, sigstore-protobuf-specs, sigstore
Successfully installed annotated-types-0.7.0 appdirs-1.4.4 betterproto-2.0.0b6 cffi-1.17.1 charset-normalizer-3.4.0 cryptography-44.0.0 dnspython-2.7.0 email-validator-2.2.0 grpclib-0.4.7 h2-4.1.0 hpack-4.0.0 hyperframe-6.0.1 id-1.5.0 importlib_resources-5.13.0 markdown-it-py-3.0.0 mdurl-0.1.2 multidict-6.1.0 pyOpenSSL-24.3.0 pycparser-2.22 pydantic-2.10.4 pydantic-core-2.27.2 pygments-2.18.0 python-dateutil-2.9.0.post0 requests-2.32.3 rich-13.9.4 securesystemslib-0.31.0 sigstore-2.1.5 sigstore-protobuf-specs-0.3.2 sigstore-rekor-types-0.0.11 tuf-3.1.1 typing-extensions-4.12.2
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/runner/.local/lib/python3.10/site-packages/sigstore/__main__.py", line 20, in <module>
    from sigstore._cli import main
  File "/home/runner/.local/lib/python3.10/site-packages/sigstore/_cli.py", line 32, in <module>
    from sigstore._internal.fulcio.client import (
  File "/home/runner/.local/lib/python3.10/site-packages/sigstore/_internal/fulcio/__init__.py", line 20, in <module>
    from .client import (
  File "/home/runner/.local/lib/python3.10/site-packages/sigstore/_internal/fulcio/client.py", line 166, in <module>
    SignedCertificateTimestamp.register(DetachedFulcioSCT)
AttributeError: type object 'cryptography.hazmat.bindings._rust.x509.Sct' has no attribute 'register'
```

See https://github.com/StevenLooman/async_upnp_client/actions/runs/12455099739/job/34767250417 for a run with this error.

Presumably, this is caused by this commit: https://github.com/pyca/cryptography/commit/4c72f368234e60a06e4a0beaf87be55940dd49c1#diff-67973ea65c939c3e29a6e4fa070ba4fadbbec96d96a94506146baf4334d618d0L39 in the cryptography package. The installed package version in the mentioned run is cryptography-44.0.0.

**Version**

v.2.1.1

**Comment by woodruffw** [comment]
Hi @StevenLooman, thanks for the report. Could you try upgrading to `v3.0.0` and seeing if the error persists? `v2.1.1` is a pretty old release, and we've released a number of dependency fixes since then.

(Out of curiosity, where did you get `v2.1.1` from? There might be an old resource we can update.)

**Comment by StevenLooman** [comment]
Thank you for getting back to me so quickly.

> Hi @StevenLooman, thanks for the report. Could you try upgrading to `v3.0.0` and seeing if the error persists? `v2.1.1` is a pretty old release, and we've released a number of dependency fixes since then.

I have bumped the version after the todays release/tagging of 0.42.0 of async-upnp-client. After failing, I created a Github release manually.

There currently is no need for a new release, and I don't want to create confusion by creating an "empty" release. If you feel that v3.0.0 fixes this, feel free to close this issue. As stated, I have since bumped the action reference to v3.0.0.

> (Out of curiosity, where did you get `v2.1.1` from? There might be an old resource we can update.)

It was from https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/ which seems to use v3.0.0 now. I have not upgraded the action reference since I have implemented this guide.
